azure-auth-helper/src/main/java/com/microsoft/azure/auth/configuration/AuthType.java

```diff
@@ -6,13 +6,92 @@
+                return AzureAuthHelper.getAzureCLICredential();
+            } catch (IOException e) {
+                throw new AzureLoginFailureException(e.getMessage());
+            }
+        }
+    },
+    DEVICE_LOGIN {
+        @Override
+        public AzureTokenWrapper getAzureToken(AuthConfiguration configuration, AzureEnvironment environment)
+                throws AzureLoginFailureException {
+            try {
+                final AzureCredential azureCredential = AzureAuthHelper.deviceLogin(environment);
+                AzureAuthHelper.writeAzureCredentials(azureCredential, AzureAuthHelper.getAzureSecretFile());
+                return new AzureTokenWrapper(DEVICE_LOGIN, AzureAuthHelper.getMavenAzureLoginCredentials(azureCredential, environment));
+            } catch (Exception e) {
+                throw new AzureLoginFailureException(e.getMessage());
+            }
+        }
+    },
+    MSI {
+        @Override
+        public AzureTokenWrapper getAzureToken(AuthConfiguration configuration, AzureEnvironment environment) {
+            return AzureAuthHelper.getMSICredential();
+        }
+    },
+    OAUTH {
+        @Override
+        public AzureTokenWrapper getAzureToken(AuthConfiguration configuration, AzureEnvironment environment)
+                throws AzureLoginFailureException {
+            try {
+                final AzureCredential azureCredential = AzureAuthHelper.oAuthLogin(environment);
+                AzureAuthHelper.writeAzureCredentials(azureCredential, AzureAuthHelper.getAzureSecretFile());
+                return new AzureTokenWrapper(OAUTH, AzureAuthHelper.getMavenAzureLoginCredentials(azureCredential, environment));
+            } catch (Exception e) {
+                throw new AzureLoginFailureException(e.getMessage());
+            }
+        }
+    },
+    SECRET_FILE {
+        @Override
+        public AzureTokenWrapper getAzureToken(AuthConfiguration configuration, AzureEnvironment environment) throws AzureLoginFailureException {
+            try {
+                return AzureAuthHelper.getSecretFileCredential();
+            } catch (IOException e) {
+                throw new AzureLoginFailureException(e.getMessage());
+            }
+        }
+    },
+    SERVICE_PRINCIPAL {
+        @Override
+        public AzureTokenWrapper getAzureToken(AuthConfiguration configuration, AzureEnvironment environment) throws AzureLoginFailureException {
+            try {
+                return AzureAuthHelper.getServicePrincipalCredential(configuration);
+            } catch (Exception e) {
+                throw new AzureLoginFailureException(e.getMessage());
+            }
+        }
+    },
+    UNKNOWN {
+        @Override
+        public AzureTokenWrapper getAzureToken(AuthConfiguration configuration, AzureEnvironment environment) {
+            return null;
+        }
+    };
+
+    public static AuthType fromString(String input) {
+        return StringUtils.isEmpty(input) ? UNKNOWN :
```

Comment:
don't map user input string to enum, user will not input 'unknown' as auth-type, for bad values, you should log a warn message and then take the default order.

Reply:
Added validation and will prompt warning for wrong auth type